services/analyzer/sequelize-tables-analyzer.js

```diff
@@ -79,52 +79,138 @@ function hasIdColumn(fields, primaryKeys) {
+function createReference(foreignKey, association, manyToManyForeignKey) {
+  const reference = {
+    foreignKey: foreignKey.columnName,
+    foreignKeyName: _.camelCase(foreignKey.columnName),
+    association,
+  };
+
+  if (association === 'belongsTo') {
+    reference.ref = foreignKey.foreignTableName;
+    reference.as = formatAliasName(foreignKey.columnName);
+  } else if (association === 'belongsToMany') {
+    reference.ref = manyToManyForeignKey.foreignTableName;
+    reference.otherKey = manyToManyForeignKey.columnName;
+    reference.junctionTable = foreignKey.tableName;
+  } else {
+    reference.ref = foreignKey.tableName;
+  }
+
+  // NOTICE: If the foreign key name and alias are the same, Sequelize will crash, we need
+  //         to handle this specific scenario generating a different foreign key name.
+  if (reference.foreignKeyName && reference.foreignKeyName === reference.as) {
+    reference.foreignKeyName = `${reference.foreignKeyName}Key`;
+  }
+
+  if (foreignKey.foreignColumnName !== 'id') {
+    reference.targetKey = foreignKey.foreignColumnName;
+  }
+
+  return reference;
+}
+
 async function analyzeTable(table, config) {
   const schema = await analyzeFields(table, config);
 
   return {
     schema,
-    foreignKeys: await tableConstraintsGetter.perform(table),
+    constraints: await tableConstraintsGetter.perform(table),
     primaryKeys: await analyzePrimaryKeys(schema),
   };
 }
 
+// NOTICE: Use the foreign key and reference properties to determine the associations
+//         and push them as references of the table.
+function defineAssociationType(databaseSchema) {
+  Object.values(databaseSchema).forEach((table) => {
+    const { constraints, primaryKeys } = table;
+    constraints.forEach((constraint) => {
+      const { columnName } = constraint;
+      const uniqueIndexes = constraint.uniqueIndexes || null;
+
+      if (constraint.columnType === 'FOREIGN KEY') {
+        const isInCompositeKey = primaryKeys
+          && primaryKeys.length > 1
+          && primaryKeys.includes(columnName);
+
+        const { isPrimary, isUnique } = checkUnicity(primaryKeys, uniqueIndexes, columnName);
+
+        const referenceTableName = constraint.foreignTableName;
+        const referenceColumnName = constraint.foreignColumnName;
+        let isManyToMany = false;
+
+        if (isInCompositeKey) {
+          // Check if the foreignKey is in a composite primary key
+          primaryKeys.forEach((primaryKey) => {
+            if (primaryKey === constraint.columnName) {
+              const manyToManyKeys = _.filter(constraints, (otherKey) =>
+                otherKey.columnName !== constraint.columnName
+                  && otherKey.columnType === 'FOREIGN KEY' && primaryKeys.includes(otherKey.columnName)) || [];
+
+              manyToManyKeys.forEach((manyToManyKey) => {
+                databaseSchema[referenceTableName].references.push(createReference(constraint, 'belongsToMany', manyToManyKey));
```

Comment:
BelongsToMany detection is very picky.
With this join table it should work in my opinion but it does not.
<img width="1712" alt="Screenshot 2019-12-17 at 09 28 46" src="https://user-images.githubusercontent.com/664261/70978296-d00b7980-20af-11ea-9bc1-f1641ad53c19.png">